code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/BasicTemplateNotificationBuilder.kt

```diff
@@ -0,0 +1,340 @@
+        tag: String?,
+        stickyNotification: Boolean
+    ) {
+        val actionButtons: List<AEPPushTemplate.ActionButton>? =
+            getActionButtonsFromString(actionButtonsString)
+        if (actionButtons.isNullOrEmpty()) {
+            return
+        }
+        for (eachButton in actionButtons) {
+            val pendingIntent: PendingIntent? =
+                if (eachButton.type === AEPPushTemplate.ActionType.DEEPLINK ||
+                    eachButton.type === AEPPushTemplate.ActionType.WEBURL
+                ) {
+                    BasicTemplateNotificationBuilder.createPendingIntent(
+                        context,
+                        trackerActivityName,
+                        eachButton.link,
+                        eachButton.label,
+                        tag,
+                        stickyNotification
+                    )
+                } else {
+                    BasicTemplateNotificationBuilder.createPendingIntent(
+                        context,
+                        trackerActivityName,
+                        null,
+                        eachButton.label,
+                        tag,
+                        stickyNotification
+                    )
+                }
+            builder.addAction(0, eachButton.label, pendingIntent)
+        }
+    }
+
+    private fun getActionButtonsFromString(actionButtons: String?): List<AEPPushTemplate.ActionButton>? {
+        if (actionButtons == null) {
+            Log.debug(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Exception in converting actionButtons json string to json object, Error :" +
+                    " actionButtons is null"
+            )
+            return null
+        }
+        val actionButtonList = mutableListOf<AEPPushTemplate.ActionButton>()
+        try {
+            val jsonArray = JSONArray(actionButtons)
+            for (i in 0 until jsonArray.length()) {
+                val jsonObject = jsonArray.getJSONObject(i)
+                val button = getActionButton(jsonObject) ?: continue
+                actionButtonList.add(button)
+            }
+        } catch (e: JSONException) {
+            Log.warning(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Exception in converting actionButtons json string to json object, Error : ${e.localizedMessage}"
+            )
+            return null
+        }
+        return actionButtonList
+    }
+
+    private fun getActionButton(jsonObject: JSONObject): AEPPushTemplate.ActionButton? {
+        return try {
+            val label = jsonObject.getString(AEPPushTemplate.ActionButtons.LABEL)
+            if (label.isEmpty()) {
+                Log.debug(PushTemplateConstants.LOG_TAG, SELF_TAG, "Label is empty")
+                return null
+            }
+            var uri: String? = null
+            val type = jsonObject.getString(AEPPushTemplate.ActionButtons.TYPE)
```

Comment:
Do we need a separate `ActionButtonType` ? Can we use `ActionType.DEEPLINK.toString()` instead?